packages/wallets/fractal/src/adapter.ts

```diff
@@ -0,0 +1,209 @@
+import base58 from 'bs58';
+import {
+    SendTransactionOptions,
+    TransactionOrVersionedTransaction,
+    WalletError,
+    WalletNotConnectedError,
+    WalletSignTransactionError,
+} from '@solana/wallet-adapter-base';
+import type { WalletName } from '@solana/wallet-adapter-base';
+import {
+    BaseSignerWalletAdapter,
+    WalletReadyState,
+    WalletConnectionError,
+    WalletPublicKeyError,
+    WalletNotReadyError,
+} from '@solana/wallet-adapter-base';
+import { Message, Transaction } from '@solana/web3.js';
+import { PublicKey } from '@solana/web3.js';
+import {
+    ConnectionManager,
+    Platform,
+    PopupEvent,
+    assertPayloadIsSolanaWalletAdapterApproved,
+    DEFAULT_POPUP_HEIGHT_PX,
+    assertPayloadIsTransactionSignatureNeededResponsePayload,
+} from '@fractalwagmi/popup-connection';
+import type { TransactionSignatureNeededPayload } from '@fractalwagmi/popup-connection';
+import uuid from 'uuid';
+
+const FRACTAL_DOMAIN_HTTPS = 'https://fractal.is';
+const APPROVE_PAGE_URL = `${FRACTAL_DOMAIN_HTTPS}/wallet-adapter/approve`;
+const SIGN_PAGE_URL = `${FRACTAL_DOMAIN_HTTPS}/wallet-adapter/sign`;
+const MIN_POPUP_HEIGHT_PX = DEFAULT_POPUP_HEIGHT_PX;
+const MAX_POPUP_WIDTH_PX = 850;
+
+export const FractalWalletName = 'Fractal Wallet' as WalletName<'Fractal Wallet'>;
+
+export class FractalWalletAdapter extends BaseSignerWalletAdapter {
+    name = FractalWalletName;
+    url = 'https://github.com/solana-labs/wallet-adapter#usage';
```

Comment:
Probably want this to be `https://fractal.is` or even better, a page specific to the wallet that tells users the benefits of it.

Reply:
Done. (The page is still being worked on on our end, but I've updated the URL preemptively.)